core/src/main/java/org/apache/calcite/sql/SqlBlobTypeNameSpec.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql;
+
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.sql.validate.SqlValidator;
+import org.apache.calcite.util.Litmus;
+
+import static org.apache.calcite.util.Static.RESOURCE;
+
+/**
+ * A SQL type name specification for the BLOB data type.
+ */
+public class SqlBlobTypeNameSpec extends SqlTypeNameSpec {
+
+  public final SqlLiteral maxLength;
+  public final SqlLobUnitSize unitSize;
+
+  /**
+   * Create a {@code SqlBlobTypeNameSpec} instance.
+   *
+   * @param maxLength The number of bytes to allocate for BLOB column
+   * @param unitSize The unit size of maxLength
+   * @param pos The parser position
+   */
+  public SqlBlobTypeNameSpec(SqlLiteral maxLength, SqlLobUnitSize unitSize,
+      SqlParserPos pos) {
+    super(new SqlIdentifier("BLOB", pos), pos);
+    this.maxLength = maxLength;
+    this.unitSize = unitSize;
+    if (maxLength != null && !isValidMaxLength(maxLength, unitSize)) {
+      throw SqlUtil.newContextException(maxLength.getParserPosition(),
+        RESOURCE.numberLiteralOutOfRange(String.valueOf(maxLength)));
+    }
+  }
+
+  private boolean isValidMaxLength(SqlLiteral maxLength,
+      SqlLobUnitSize unitSize) {
+    int numericMaxLength = maxLength.getValueAs(Integer.class);
+    switch (unitSize) {
+    case UNSPECIFIED:
+      if (numericMaxLength > 2097088000) {
```

Comment:
Add a comment on what the assumed unitSize is if it is unspecified?

Reply:
Sure, I could add a comment in the enum for unspecified saying it would default to bytes

Reply:
I think you're still missing this comment?

Reply:
I have it clarified in the [enum](https://github.com/googleinterns/calcite/blob/bde051697145e5651985547fb7ea78fb98694276/core/src/main/java/org/apache/calcite/sql/SqlLobUnitSize.java#L36), but if you you thinks it's needed I could also comment that here